File: Day2.java
import java.nio.file.Paths;
import java.nio.file.Files;
import java.util.List;

public class Day2 {

    public static String formatOpponentChoice(String oppChoice) {
        switch (oppChoice) {
            case "A":
                return "rock";
            case "B":
                return "paper";
            default:
                return "scissors";
        }
    }

    public static String formatYourChoice(String oppChoice) {
        switch (oppChoice) {
            case "X":
                return "rock";
            case "Y":
                return "paper";
            default:
                return "scissors";
        }
    }

    public static String formatYourChoice2(String oppChoice) {
        switch (oppChoice) {
            case "X":
                return "lose";
            case "Y":
                return "draw";
            default:
                return "win";
        }
    }

    public static void main(String[] args) throws Exception {
        List<String> lines = Files.readAllLines(Paths.get("input2.txt"));
        int score = 0;
        int score2 = 0;

        for (int i = 0; i < lines.size(); i++) {
            String line = lines.get(i).trim();
            String oppChoice = formatOpponentChoice(line.split(" ")[0]);
            String yourChoice = formatYourChoice(line.split(" ")[1]);
            String yourChoice2 = formatYourChoice2(line.split(" ")[1]);
            switch (oppChoice) {
                case "rock":
                    if (yourChoice == "rock")
                        score += 3 + 1;
                    else if (yourChoice == "paper")
                        score += 6 + 2;
                    else
                        score += 0 + 3;

                    if (yourChoice2 == "draw")
                        score2 += 3 + 1;
                    else if (yourChoice2 == "win")
                        score2 += 6 + 2;
                    else
                        score2 += 0 + 3;

                    break;
                case "paper":
                    if (yourChoice == "rock")
                        score += 0 + 1;
                    else if (yourChoice == "paper")
                        score += 3 + 2;
                    else
                        score += 6 + 3;

                    if (yourChoice2 == "lose")
                        score2 += 0 + 1;
                    else if (yourChoice2 == "draw")
                        score2 += 3 + 2;
                    else
                        score2 += 6 + 3;
                    break;

                case "scissors":
                    if (yourChoice == "rock")
                        score += 6 + 1;
                    else if (yourChoice == "paper")
                        score += 0 + 2;
                    else
                        score += 3 + 3;

                    if (yourChoice2 == "win")
                        score2 += 6 + 1;
                    else if (yourChoice2 == "lose")
                        score2 += 0 + 2;
                    else
                        score2 += 3 + 3;
                    break;
            }
        }

        System.out.println("Score: " + score);
        System.out.println("Score2: " + score2);
    }
}

File: day2.ts
import { file } from "./input";
import colors from "colors";
import { daynumber } from "./constants";

function getValueForSymbol(symbol: string) {
  switch (symbol) {
    case "A":
      return "rock";
    case "B":
      return "paper";
    case "C":
      return "scissors";
    case "Y":
      return "rock";
    case "X":
      return "paper";
    case "Z":
      return "scissors";
  }

  return "";
}

function getValueForSymbol2(symbol: string) {
  switch (symbol) {
    case "A":
      return "rock";
    case "B":
      return "paper";
    case "C":
      return "scissors";
    case "Y":
      return "draw";
    case "X":
      return "lose";
    case "Z":
      return "win";
  }

  return "";
}

function getOutcomeScore(p: string, o: string) {
  if (o == "rock") {
    if (p == "rock") return 3 + 1;
    else if (p == "paper") return 6 + 2;
    else return 0 + 3;
  } else if (o == "paper") {
    if (p == "paper") return 3 + 2;
    else if (p == "scissors") return 6 + 3;
    else return 0 + 1;
  } else {
    if (p == "scissors") return 3 + 3;
    else if (p == "rock") return 6 + 1;
    else return 0 + 2;
  }
}

function getOutcomeScore2(p: string, o: string) {
  if (o == "rock") {
    if (p == "draw") return 3 + 1;
    else if (p == "win") return 6 + 2;
    else return 0 + 3;
  } else if (o == "paper") {
    if (p == "draw") return 3 + 2;
    else if (p == "win") return 6 + 3;
    else return 0 + 1;
  } else {
    if (p == "draw") return 3 + 3;
    else if (p == "win") return 6 + 1;
    else return 0 + 2;
  }
}

function calculate(input: string) {
  return input.split("\n").map((line) => {
    const symbols = line.trim().split(" ");

    const [oppChoice, playerChoice] = symbols.map((symbol) =>
      getValueForSymbol(symbol)
    );
    const [, needOutcome] = symbols.map((symbol) => getValueForSymbol2(symbol));

    return {
      outcome: getOutcomeScore(playerChoice, oppChoice),
      aleternativeOutcome: getOutcomeScore2(needOutcome, oppChoice),
    };
  });
}

export function execute() {
  console.log(colors.green("Read input for") + ` day ${daynumber}`.yellow);
  const strategyGuide = calculate(file);

  let totalScore = strategyGuide.reduce((acc, item) => acc + item.outcome, 0);

  console.log("The solution #1 is ".green + ` ${totalScore} points`.yellow);

  totalScore = strategyGuide.reduce(
    (acc, item) => acc + item.aleternativeOutcome,
    0
  );

  console.log("The solution #2 is ".green + ` ${totalScore} points`.yellow);
}
